libeos-payg/clock.c

```diff
@@ -0,0 +1,90 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Alternatively, the contents of this file may be used under the terms of the
+ * GNU Lesser General Public License Version 2.1 or later (the "LGPL"), in
+ * which case the provisions of the LGPL are applicable instead of those above.
+ * If you wish to allow use of your version of this file only under the terms
+ * of the LGPL, and not to allow others to use your version of this file under
+ * the terms of the MPL, indicate your decision by deleting the provisions
+ * above and replace them with the notice and other provisions required by the
+ * LGPL. If you do not delete the provisions above, a recipient may use your
+ * version of this file under the terms of either the MPL or the LGPL.
+ */
+
+#include <clock.h>
+#include <glib.h>
+
+G_DEFINE_INTERFACE (EpgClock, epg_clock, G_TYPE_OBJECT)
+
+static void
+epg_clock_default_init (EpgClockInterface *iface)
+{
+}
+
+/**
+ * epg_clock_get_wallclock_time:
+ * @self: an #EpgClock
+ *
+ * Like g_get_real_time(), this returns the wall clock time, but the
+ * implementation determines what type of clock to use.
+ *
+ * Returns: the number of microseconds since January 1, 1970 UTC
+ **/
```

Comment:
Nitpick: I can’t be bothered to file suggestions for all of these, but they all need `Since: 0.2.1` and to use `*/` to terminate, rather than `**/` (the former is the gtk-doc preferred style).